contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -49,7 +49,14 @@
 
 
 debug = False
-__names = set()
+
+# If enabled, output recursive traversal info of _transcribe() method.
+transcribe_debug = False
+
+
+def transcribe_log(s, indent=0, alwaysPrint=False):
```

Comment:
This should have an `_` prefix since it is for internal use only.

Reply:
ok.

Reply:
(But I followed the debug flag just right above, which is also internal as far as I can see. That should be _debug then as well?)

Reply:
We've tried to follow that convention for functions. (e.g. `_transcode`) Maybe we should do that for module-level variables as well? @ssteinbach do you have an opinion?

You could argue that the toggles to turn this on/off are useful from outside the module - e.g. when debugging an application that uses this module.

Reply:
Yeah.  @meshula we should add these notes to the in progress style guide.  Some stuff:

- I would vote to remove the `transcribe_debug` variable entirely (and any if statements that were only switched by that)
- if its going to exist it should be all caps (module level constant) + prefixed by a `_` (not intended to pollute the namespace)
- I think josh was commenting on the `transcribe` function, which if its for internal usage should also be prefixed by a `_`, `_transcribe()` to not pollute the namespace and signal its intent.
- finally, we're using snake case a la pep8, so the argument should be `always_print` rather than `alwaysPrint`.

see also: https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

> Function and Variable Names
Function names should be lowercase, with words separated by underscores as necessary to improve readability.
>
> Variable names follow the same convention as function names.
>
> mixedCase is allowed only in contexts where that's already the prevailing style (e.g. threading.py), to retain backwards compatibility.


Also:
>Constants
>Constants are usually defined on a module level and written in all capital letters with underscores separating words. Examples include MAX_OVERFLOW and TOTAL

Reply:
Just to clarify, I prefer to remove debug code - it decreases test coverage, usually requires special knowledge to turn on, and for python at least is rarely as useful as using a dedicated debugger like `pdb`.  Unless its exposed as top level interface functionality to users.  Open to counter point on that one though; what do the others think?

Reply:
I am sorry, but I disagree in this point strongly. The AAF-object-structure is hard to understand and you need some debug tools in place to manage the complexity of _transcribe(). I actually use this flag in conjunction with pdb. I activate the flag and then do a set_trace() in the part of the traversal-tree I am interested in. So If I set it in SourceClip, I can see right away what traversal happened until this point, giving me some important context of how to introspect my current state in relation to the traversal.

Reply:
Also just to clarify, I agree with your point in most (almost all) cases, but there are some exceptions where a complex, 'deep' structure requires some debug-bandaid-code put in place, so whoever wants to do a deep dive into that code has already something in place and does not require to scaffold the code from ground up. IMHO such situations are exceptions but should outweigh the desire of a perfect code-coverage.

Reply:
I like the idea of turning it on via an adapter argument. We will be improving this adapter for a long time to come, and this sort of diagnostic feature is akin to the simplify=False option that is already an adapter argument.

Reply:
Oh, and I guess I wasn't clear. It was mostly the function `transcribe_log()` that I was suggesting should be `_transcribe_log()`

Reply:
yes, I like the idea to have it be an adapter argument. And yes, I totally did not realize you meant the method. I'll fix that.